pybamm/callbacks.py

```diff
@@ -1,10 +1,3 @@
-#
-# Base class for callbacks and some useful callbacks for pybamm
-# Callbacks are used to perform actions (e.g. logging, saving)
-# at certain points in the simulation
-# Inspired by Keras callbacks
-# https://github.com/keras-team/keras/blob/master/keras/callbacks/callback.py
```

Comment:
Can we replace this with the updated link? It is often useful to look at the original implementations, especially for people trying to extend PyBaMM (or any software) and not just use it.

Reply:
Replacing with a GitHub permalink (pinned to a commit hash) would also be helpful, so that we can know what the implementation was at the time of writing and the link won't break.

Reply:
I don't think we need to mark were we were inspired by other projects. Especially when concepts like callbacks are pretty common.

We should use links only where it is necessary. The fact is that this keras link broke just 6 months ago too and this time it was not appearing in the link checker log until the stackoverflow stuff was removed.